src/warc2zim/converter.py

```diff
@@ -421,42 +427,47 @@ def find_icon_and_language(self, content):
                 return
 
     def retrieve_illustration(self):
-        """sets self.illustration from self.favicon_url either from WARC or download
+        """sets self.illustration either from WARC or download
 
         Uses fallback in case of errors/missing"""
-        if not self.favicon_url:
-            self.favicon_url = "fallback.png"
-            self.illustration = DEFAULT_DEV_ZIM_METADATA["Illustration_48x48_at_1"]
-            return
-        # look into WARC records first
-        for record in self.iter_all_warc_records():
-            url = get_record_url(record)
-            if not url or record.rec_type == "revisit":
-                continue
-            if url == self.favicon_url:
-                logger.debug(f"Found WARC record for favicon: {self.favicon_url}")
-                if record and record.http_headers.get_statuscode() != "200":
-                    logger.warning("WARC record for favicon is unuable. Skipping")
-                    self.favicon_url = "fallback.png"
-                    self.illustration = DEFAULT_DEV_ZIM_METADATA[
-                        "Illustration_48x48_at_1"
-                    ]
-                    return
-                self.illustration = get_record_content(record)
-
-        # favicon_url not in WARC ; downloading
-        try:
-            dst = io.BytesIO()
-            if not stream_file(self.favicon_url, byte_stream=dst)[0]:
-                raise OSError(
-                    "No bytes received downloading favicon"
-                )  # pragma: no cover
-            self.illustration = dst.getvalue()
-        except Exception as exc:
-            logger.warning(f"Unable to retrieve favicon. Using fallback: {exc}")
-            self.favicon_url = "fallback.png"
+
+        if self.favicon_url or self.favicon_path:
+            # look into WARC records
+            for record in self.iter_all_warc_records():
+                if record.rec_type != "response":
+                    continue
+                url = get_record_url(record)
+                path = normalize(url)
+                if path == self.favicon_path or url == self.favicon_url:
+                    logger.debug("Found WARC record for favicon")
+                    if (
+                        record.http_headers
+                        and record.http_headers.get_statuscode() != "200"
+                    ):  # pragma: no cover
+                        logger.warning("WARC record for favicon is unusable")
+                        break
+                    self.illustration = get_record_content(record)
+                    break
```

Comment:
Should we have a `return` or something like that to not download the favicon just after ?

Reply:
Good point, I fixed it slightly differently since I do want to `del` the variables.